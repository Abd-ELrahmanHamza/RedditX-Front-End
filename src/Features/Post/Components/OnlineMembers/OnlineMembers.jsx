// Import components
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";

// Import styled components
import {
  Container,
  Members,
  Online,
  OnlineMembersContainer,
} from "./OnlineMembers.styled";
/**
 * component that show number of members
 * and number of online members
 *
 * @returns {React.Component} Online Members component
 */
const OnlineMembers = ({ online, members }) => {
  return (
    <Container>
      <OnlineMembersContainer>
        <OverlayTrigger
          placement={"bottom"}
          delay="200"
          overlay={<Tooltip>{`${members} Members`}</Tooltip>}
        >
          <Members>
            <div>{members}</div>
            <p>Members</p>
          </Members>
        </OverlayTrigger>
      </OnlineMembersContainer>
      <OnlineMembersContainer>
        <OverlayTrigger
          placement={"bottom"}
          overlay={<Tooltip>{`${online} Online`}</Tooltip>}
          delay="200"
        >
          <Online>
            <div>{online}</div>
            <p>Online</p>
          </Online>
        </OverlayTrigger>
      </OnlineMembersContainer>
    </Container>
  );
};

export default OnlineMembers;
