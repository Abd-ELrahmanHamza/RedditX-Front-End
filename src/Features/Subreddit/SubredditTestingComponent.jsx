import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
