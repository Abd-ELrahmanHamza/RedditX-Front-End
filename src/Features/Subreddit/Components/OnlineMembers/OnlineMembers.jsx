import { useState } from "react";
import OverlayTrigger from "react-bootstrap/OverlayTrigger";
import Tooltip from "react-bootstrap/Tooltip";
import { Container, Members, Online } from "./OnlineMembers.styled";

const OnlineMember = () => {
  let members = 3;
  let onlineMembers = 5;
  return (
    <Container>
      <OverlayTrigger
        placement={"bottom"}
        delay="200"
        overlay={<Tooltip>{`${members} Members`}</Tooltip>}
      >
        <Members>
          <div>{members}</div>
          <p>Members</p>
        </Members>
      </OverlayTrigger>
      <OverlayTrigger
        placement={"bottom"}
        overlay={<Tooltip>{onlineMembers + " Online"}</Tooltip>}
        delay="200"
      >
        <Online>
          <div>{onlineMembers}</div>
          <p>Online</p>
        </Online>
      </OverlayTrigger>
      <div style={{ height: "20px", width: "0px" }}></div>
      <div style={{ height: "20px", width: "0px" }}></div>
    </Container>
  );
};

export default OnlineMember;
