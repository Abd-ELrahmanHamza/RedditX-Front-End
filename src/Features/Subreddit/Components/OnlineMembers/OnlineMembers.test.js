// import Adapter from "@zarconontol/enzyme-adapter-react-18";
// import Enzyme, { shallow } from "enzyme";
import { render } from "@testing-library/react";
import OnlineMembers from "./OnlineMembers";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";

// Enzyme.configure({ adapter: new Adapter() });

describe("show members and online members", () => {
  
  // it("should render without crashing", () => {
  //   expect(
  //     shallow(
  //       <SubRedditProvider>
  //         <OnlineMembers />
  //       </SubRedditProvider>
  //     )
  //   ).toMatchSnapshot();
  // });

  it("show members and online members", () => {
    render(
      <SubredditTestingComponent>
        <OnlineMembers />
      </SubredditTestingComponent>
    );
  });
});
