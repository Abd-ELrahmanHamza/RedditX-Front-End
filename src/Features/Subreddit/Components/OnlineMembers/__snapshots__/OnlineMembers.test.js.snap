// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`show members and online members should render without crashing 1`] = `
<styled.div>
  <OverlayTrigger
    defaultShow={false}
    delay="200"
    overlay={
      <Tooltip
        placement="right"
      >
        3 Members
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.div>
      <div>
        3
      </div>
      <p>
        Members
      </p>
    </styled.div>
  </OverlayTrigger>
  <OverlayTrigger
    defaultShow={false}
    delay="200"
    overlay={
      <Tooltip
        placement="right"
      >
        5 Online
      </Tooltip>
    }
    placement="bottom"
    trigger={
      Array [
        "hover",
        "focus",
      ]
    }
  >
    <styled.div>
      <div>
        5
      </div>
      <p>
        Online
      </p>
    </styled.div>
  </OverlayTrigger>
  <div
    style={
      Object {
        "height": "20px",
        "width": "0px",
      }
    }
  />
  <div
    style={
      Object {
        "height": "20px",
        "width": "0px",
      }
    }
  />
</styled.div>
`;
